src/Umbraco.Core/Services/ContentEditingServiceBase.cs

```diff
@@ -23,29 +23,43 @@ public abstract class ContentEditingServiceBase<TContent, TContentType, TContent
 
     private readonly ICoreScopeProvider _scopeProvider;
 
+    private readonly ITreeEntitySortingService _treeEntitySortingService;
+
     protected ContentEditingServiceBase(
         TContentService contentService,
         TContentTypeService contentTypeService,
         PropertyEditorCollection propertyEditorCollection,
         IDataTypeService dataTypeService,
         ILogger<ContentEditingServiceBase<TContent, TContentType, TContentService, TContentTypeService>> logger,
-        ICoreScopeProvider scopeProvider)
+        ICoreScopeProvider scopeProvider,
+        ITreeEntitySortingService treeEntitySortingService)
     {
         _propertyEditorCollection = propertyEditorCollection;
         _dataTypeService = dataTypeService;
         _logger = logger;
         _scopeProvider = scopeProvider;
+        _treeEntitySortingService = treeEntitySortingService;
         ContentService = contentService;
         ContentTypeService = contentTypeService;
     }
 
-    protected abstract TContent Create(string? name, int parentId, TContentType contentType);
+    protected delegate TContent Create(string? name, int parentId, TContentType contentType);
+
+    protected delegate OperationResult? Move(TContent content, int newParentId);
+
+    protected delegate TContent? Copy(TContent content, int newParentId);
+
+    protected delegate OperationResult? Delete(TContent content);
+
+    protected delegate IEnumerable<TContent> GetPagedChildren(int parentId, int pageIndex, int pageSize, out long total);
+
+    protected delegate ContentEditingOperationStatus Sort(IEnumerable<TContent> items);
```

Comment:
Does this have to be delegates? Couldn't this be abstract methods instead, with overwritten implementations in the respective implementations 🤔

Reply:
Yeah... I have changed that now.

Having them as delegates allowed us to act on the context in which they were being executed. And they also did not force the media editing service to implement copy, which is not supported. But I do like abstracts a whole lot better, so now they're abstracts. The upside is that we can remove some duplicate boilerplate code from the content and media editing services.

While redoing this I discovered that we were able to delete items that were not in the recycle bin 🤦 I have fixed that too.